scripts/upload_rrd.py

```diff
@@ -0,0 +1,164 @@
+        Low-level upload of data.
+
+        Parameters
+        ----------
+        data:
+            The data to upload.
+        gcs_path:
+            The path of the object.
+        content_type:
+            The content type of the object.
+        content_encoding:
+            The content encoding of the object.
+
+        """
+
+        logging.info(f"Uploading {gcs_path} (size: {len(data)}, type: {content_type}, encoding: {content_encoding})")
+        destination = self.bucket.blob(gcs_path)
+        destination.content_type = content_type
+        destination.content_encoding = content_encoding
+
+        if destination.exists():
+            logging.warn(f"blob {gcs_path} already exists in GCS, skipping upload")
+            return
+
+        stream = BytesIO(data)
+        destination.upload_from_file(stream)
+
+
+def data_hash(data: bytes) -> str:
+    """Compute a sha1 hash digest of some data."""
+    return hashlib.sha1(data).hexdigest()
+
+
+def download_file(url: str, path: Path) -> None:
+    path.parent.mkdir(parents=True, exist_ok=True)
+    logging.info("Downloading %s to %s", url, path)
+    response = requests.get(url, stream=True)
+    with tqdm.tqdm.wrapattr(
+        open(path, "wb"),
+        "write",
+        miniters=1,
+        total=int(response.headers.get("content-length", 0)),
+        desc=f"Downloading {path.name}",
+    ) as f:
+        for chunk in response.iter_content(chunk_size=4096):
+            f.write(chunk)
+
+
+DESCRIPTION = """Upload an .rrd to static.rerun.io.
+
+    pixi run upload-rrd --version 0.15.0 path/to/recording.rrd
+
+    The version is used for two things:
+    A) used as a folder name in the GCS bucket.
+    B) used to generate a link to the correct version of the Rerun web viewer.
+"""
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(description=DESCRIPTION, formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument("path", type=str, help="Recording .rrd to upload")
+    parser.add_argument(
+        "--name", type=str, required=False, help="Name of the recording. If not supplied, the file name is used."
+    )
+    parser.add_argument("--version", type=str, required=True, help="The Rerun version, e.g. '0.15.0'.")
+    parser.add_argument("--debug", action="store_true", help="Enable debug logging.")
+    args = parser.parse_args()
+
+    if args.debug:
+        logging.basicConfig(level=logging.DEBUG)
+    else:
+        logging.basicConfig(level=logging.INFO)
+
+    try:
+        if not args.path.endswith(".rrd"):
+            raise RuntimeError("File path expected to have .rrd extension")
+
+        file_path = Path(args.path)
+        name = args.name or file_path.stem
+        version = args.version
+
+        # Check if user put `v0.15.0` instead of `0.15.0`:
+        if m := re.match(r"v(\d+\.v\d+\..*)", version):
+            version = m.group(1)
+            raise RuntimeError("Version should be in the format '{version}', without a leading 'v'")
```

Comment:
This check doesn't seem to work, I've just executed this successfully:
```
pixi run upload-rrd plots.rrd --version v0.15.1
```

-> https://static.rerun.io/rrd/v0.15.1/plots_52994f4cc6bc625e2e9276ab7f97fcba65f61020.rrd
-> https://console.cloud.google.com/storage/browser/rerun-rrd/v0.15.1;tab=objects?project=rerun-open